src/data-set/display.ts

```diff
@@ -0,0 +1,233 @@
+  return `${tokenAddress.slice(0, 6)}...${tokenAddress.slice(-4)}`
+}
+
+/**
+ * Format storage price in USDFC per TiB per month
+ * Always shows TiB/month for consistency, with appropriate precision
+ */
+function formatStoragePrice(pricePerTiBPerDay: bigint): string {
+  try {
+    const priceInUSDFC = parseFloat(ethers.formatUnits(pricePerTiBPerDay, 18))
+
+    // Handle very small prices that would show as 0.0000
+    if (priceInUSDFC < 0.0001) {
+      return '< 0.0001 USDFC/TiB/day'
+    }
+
+    // For prices >= 0.0001, show with appropriate precision
+    if (priceInUSDFC >= 1) {
+      return `${priceInUSDFC.toFixed(2)} USDFC/TiB/day`
+    } else if (priceInUSDFC >= 0.01) {
+      return `${priceInUSDFC.toFixed(4)} USDFC/TiB/day`
+    } else {
+      return `${priceInUSDFC.toFixed(6)} USDFC/TiB/day`
+    }
+  } catch {
+    return pc.red('invalid price')
+  }
+}
+
+function renderNetworkDetails(network: string, address: string): void {
+  log.line(`Network: ${pc.bold(network)}`)
+  log.line(`Client address: ${address}`)
+  log.line('')
+}
+
+function renderDataSetHeader(dataSet: DataSetSummary): void {
+  log.line(`${pc.bold(`#${dataSet.dataSetId}`)}`)
+  log.indent(`Status: ${statusLabel(dataSet)}`)
+  log.indent(`CDN add-on: ${dataSet.withCDN ? 'enabled' : 'disabled'}`)
+  log.line('')
+}
+
+function renderProviderDetails(dataSet: DataSetSummary, indentLevel: number = 0): void {
+  log.indent(pc.bold('Provider'), indentLevel)
+  log.indent(`ID: ${dataSet.providerId}`, indentLevel + 1)
+  log.indent(`Address: ${dataSet.serviceProvider}`, indentLevel + 1)
+  if (dataSet.provider == null) {
+    log.line('')
+    return
+  }
+  log.indent(`Name: ${dataSet.provider.name}`, indentLevel + 1)
+  log.indent(`Description: ${dataSet.provider.description}`, indentLevel + 1)
+  log.indent(`Service URL: ${dataSet.provider.products.PDP?.data?.serviceURL ?? 'unknown'}`, indentLevel + 1)
+  log.indent(`Active: ${dataSet.provider.active ? 'yes' : 'no'}`, indentLevel + 1)
+  log.indent(
+    `Min piece size: ${formatBytes(BigInt(dataSet.provider.products.PDP?.data?.minPieceSizeInBytes ?? 0))}`,
+    indentLevel + 1
+  )
+  log.indent(
+    `Max piece size: ${formatBytes(BigInt(dataSet.provider.products.PDP?.data?.maxPieceSizeInBytes ?? 0))}`,
+    indentLevel + 1
+  )
+  log.indent(
+    `Storage price: ${formatStoragePrice(dataSet.provider.products.PDP?.data?.storagePricePerTibPerDay ?? 0n)}`,
+    indentLevel + 1
+  )
+  log.indent(
+    `Min proving period: ${dataSet.provider.products.PDP?.data?.minProvingPeriodInEpochs ?? 0} epochs`,
+    indentLevel + 1
+  )
+  log.indent(`Location: ${dataSet.provider.products.PDP?.data?.location ?? 'unknown'}`, indentLevel + 1)
+  log.indent(
+    `Payment token: ${formatPaymentToken(dataSet.provider.products.PDP?.data?.paymentTokenAddress ?? 'unknown')}`,
+    indentLevel + 1
+  )
+  log.indent(`Commission: ${formatCommission(dataSet.commissionBps)}`, indentLevel + 1)
```

Comment:
let's hide this _unless_ it's nonzero, it should always be zero for now, FWSS has 0% commission